Docs: New panel editor documentation (#24711)
grafana/grafana: docs/sources/panels/panel-editor.md

+++
title = "Panel editor"
type = "docs"
[menu.docs]
identifier = "panel-editor"
weight = 200
+++

# Panel editor

This page describes the parts of the Grafana panel editor and links to where you can find more information.

> {{< docs-imagebox img="/img/docs/panel-editor/panel-editor-7-0.png" class="docs-image--no-shadow" max-width="1500px" >}}

Reviewer: https://github.com/grafana/website/blob/60b482af9068d160eecc9d38d58fba8a1842eb78/static/static/img/docs/panel-editor/panel-editor-7-0.png 

image does not seem ideal, very high res with lots of empty space and with side panel collapsed. would use something like: 
<img width="1614" alt="Screen Shot 2020-05-14 at 19 46 45" src="https://user-images.githubusercontent.com/10999/82069228-4a7a0d00-96d3-11ea-8623-c5ec9fadd17c.png">

Author: The point of the image is to show all the parts of the UI. This one does not do that. 

I resized my image and trimmed the empty space.

Reviewer: sorry wong image

Reviewer: I meant this

<img width="1680" alt="Screen Shot 2020-05-15 at 17 40 59" src="https://user-images.githubusercontent.com/10999/82072306-cfffbc00-96d7-11ea-92fb-852f770eb301.png">

Author: Ooh, pretty! Yes, I'll add that one.

## Open the Panel editor

There are several ways to access the panel editor, also called the **Edit Panel** screen, _edit mode_, or _panel edit mode_:

- Click the **Add panel** icon at the top of the screen and then click **Add new panel**. The new panel opens in the panel editor. For detailed instructions on how to add a panel, refer to [Add a panel]({{< relref "add-a-panel.md" >}})
- Click the title of an existing panel and then click **Edit**. The panel opens in edit mode.
- Click anywhere on an existing panel and then press **e** on your keyboard. The panel opens in edit mode.

## Resize panel editor sections

> Drag to resize sections of the panel editor. If the Panel, Field, and Overrides tabs become too narrow, then the tabs convert to a dropdown list.
> 

Reviewer 2: ```suggestion
Drag to resize sections of the panel editor. If the side pane becomes too narrow, then the Panel, Field, and Overrides tabs change to a dropdown list.

```

Reviewer 3: We're using "side pane" in documentation? Just checking as so far we were referring to it as "sidebar".

Reviewer 2: Personally I prefer "side pane", think this is what we used in other docs too so far

Author: Also, "sidebar" usually refers to the icons on the left of the dashboard.

{{< docs-imagebox img="/img/docs/panel-editor/resize-panel-editor-panels-7-0.gif" class="docs-image--no-shadow" max-width="600px" >}}

## Parts of the panel editor

This section describes the parts of the panel editor screen and a bit about fields, options, or tasks associated with each part. Some sections in this page link to pages where sections or tasks are documented more fully.

### Header

The header section lists the name of the dashboard that the panel is in and some dashboard commands. You can also click the **Go back** arrow to return to the dashboard.

{{< docs-imagebox img="/img/docs/panel-editor/edit-panel-header-7-0.png" class="docs-image--no-shadow" max-width="1000px" >}}

On the right side of the header are the following options:

- **Dashboard settings (gear) icon -** Click to access the dashboard settings.

> - **Discard -** Discards all changes you have made since you last saved the dashboard.

Reviewer 2: ```suggestion
- **Discard -** Discards all changes you have made to the panel since you last saved the dashboard.
```

- **Save -** Saves the dashboard, including all changes you have made in the panel editor.

> - **Apply -** Applies changes you made and then closes the panel editor, returning you to the dashboard.

Reviewer 2: ```suggestion
- **Apply -** Applies changes you made and then closes the panel editor, returning you to the dashboard. You will have to save the dashboard to persist the applied changes.
```

> Panel title, access dashboard settings, Save/Apply/Discard

Reviewer 2: ```suggestion
```

Hmm? :)

### Visualization preview

The visualization preview section contains viewing options, time range controls, the visualization preview, and (if applicable) the panel title, axes, and legend.

{{< docs-imagebox img="/img/docs/panel-editor/visualization-preview-7-0.png" class="docs-image--no-shadow" max-width="1200px" >}}

> 

Reviewer: Not sure this section is needed. And the panel title sure is part of the visualization shown, but not sure if that needed to spelled out. And Axes and legend? Why mention a graph specific option here? 

Think we can drop the image box here.

Author: Yes, the section is needed. The point of this topic is to explain ALL the sections of the Panel editor.

Graph is not the only visualization that has axes and legends.

> - **Fill -**

Reviewer 2: ```suggestion
- **Fill -** The visualization preview will fill the available space in the preview part.
```

> - **Fill -**

Reviewer 3: ```suggestion
- **Fill -** The visualization preview will fill the available space in the preview part. If you change the width of the side pane or height of the bottom pane the visualisation will adapt to fill whatever space is available.
```

Reviewer 2: @oddlittlebird I've removed mine suggestion as I like this more :)

> - **Fit -**

Reviewer 2: ```suggestion
- **Fit -** The visualization preview will fill the available space in but preserve the aspect ratio of the panel.
```

> - **Exact -**

Reviewer 2: ```suggestion
- **Exact -** The visualization preview will have the exact size as the size on the dashboard. If not enough space is available, the visualization will scale down preserving the aspect ratio.
```

- **Time range controls -** For more information, refer to [Time range controls](ADD LINK WHEN TOPIC IS MERGED).

> ### Data section

Reviewer 3: ```suggestion
### Data section (bottom pane)
```

Reviewer 2: Yes!

> The section contains tabs where you enter queries, transform your data, and create alert rules.

Reviewer 3: ```suggestion
The section contains tabs where you enter queries, transform your data, and create alert rules (if applicable).
```

Reviewer 3: It's not always possible to create alert rules. Eg. use of transformations block this.

{{< docs-imagebox img="/img/docs/panel-editor/data-section-7-0.png" class="docs-image--no-shadow" max-width="1200px" >}}

- **Query tab -** Select your data source and enter queries here. For more information, refer to [Queries]({{< relref "queries.md" >}}).
- **Transform tab -** Apply data transformations. For more information, refer to [Transformations]({{< relref "transformations.md" >}}).
- **Alert tab -** Write alert rules. For more information, refer to [Create alerts]({{< relref "../alerting/create-alerts.md" >}}).

> ### Panel and field configuration options

Reviewer 3: ```suggestion
### Panel and field configuration options (side pane)
```

Reviewer 2: Yes!

> The section contains tabs is where you control almost every aspect of how your data is visualized. Not all tabs are available for each visualization.

Reviewer 2: ```suggestion
The section contains tabs where you control almost every aspect of how your data is visualized. Not all tabs are available for each visualization.
```
